thirdparty-api/data-models.md

```diff
@@ -0,0 +1,953 @@
+- Error Callback - `PUT /thirdpartyRequests/authorizations/<ID>/error`
+- Data Model – Empty body
+###### 3.1.6.1.2 `POST /thirdpartyRequests/authorizations`
+
+Used by: DFSP
+
+The HTTP request `POST /thirdpartyRequests/authorizations` is used to request the validation by a
+ customer for the transfer described in the request.
+
+Callback and data model information for `POST /thirdpartyRequests/authorizations:`
+
+- Callback - `PUT /thirdpartyRequests/authorizations/<ID>`
+- Error Callback - `PUT /thirdpartyRequests/authorizations/<ID>/error`
+- Data Model – See Table below
+
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| authorizationRequestId | 1 | CorrelationId | Common ID between the PISP and the Payer DFSP for the authoriztion request object. The ID should be reused for resends of the same authoriztion request. A new ID should be generated for each new authoriztion request. |
+| transactionRequestId   | 1 | CorrelationId | The unique identifier of the transaction request for which authorization is being requested. |
+| challenge              | 1 | BinaryString | The challenge that the PISP’s client is to sign. |
+| transferAmount         | 1 | Money | The amount that will be debited from the sending customer’s account as a consequence of the transaction. |
+| payeeReceiveAmount     | 1 | Money | The amount that will be credited to the receiving customer’s account as a consequence of the transaction. |
+| fees                   | 1 | Money | The amount of fees that the paying customer will be charged as part of the transaction. |
+| payer                  | 1 | PartyIdInfo | Information about the Payer type, id, sub-type/id, FSP Id in the proposed financial transaction. |
+| payee                  | 1 | Party | Information about the Payee in the proposed transaction |
+| transactionType        | 1 | TransactionType | The type of the transaction. |
+| expiration             | 1 | DateTime | The time by which the transfer must be completed, set by the payee DFSP. |
+| extensionList          | 0..1 | ExtensionList |Optional extension, specific to deployment. |
+
+##### 3.1.6.2 Callbacks
+The following callbacks are supported for the `/thirdpartyRequests/authorizations` resource
+###### 3.1.6.2.1 `PUT /thirdpartyRequests/authorizations/<ID>`
+
+Used by: PISP
+
+After receiving the `POST /thirdpartyRequests/authorizations`, the PISP will present the details of the 
+transaction to their user, and request that the client sign the `challenge` field using the credential
+they previously registered.
+
+The signed challenge will be sent back by the PISP in `PUT /thirdpartyRequests/authorizations/<ID>`:
+
+| Name | Cardinality | Type | Description |
+| --- | --- | --- | --- |
+| responseType  | 1    | AuthorizationResponseType | `ACCEPTED` or `REJECTED` |
+| signedPayload | 0..1 | SignedPayload | If the `responseType` is `ACCEPTED`, `signedPayload` is required. |
+
+
+`SignedPayload` is defined as follows:
```

Comment:
> `SignedPayload` is defined as follows:
This should be defined under Section 3.2, not as part of the definition of the callback.